src/exm-browse-page.c

```diff
@@ -184,6 +185,21 @@ on_first_page_result (GObject       *source,
 
     to_append = exm_search_provider_query_finish (EXM_SEARCH_PROVIDER (source), res, &self->max_pages, &error);
 
+    if (error)
+    {
+        // Filter 5xx status codes (server errors)
+        if (error->code / 100 == 5)
+            gtk_label_set_markup (self->error_label, _("Check <a href='https://status.gnome.org/'>GNOME infrastructure</a> status and try again"));
+        else
+            gtk_label_set_text (self->error_label, _("Check your network status and try again"));
```

Comment:
Should we return these error messages in `error->message` from `exm-request-handler` instead? The messages are repeated for all three views, although maybe it is a message more intended to be displayed in the UI.